judge/models/profile.py

```diff
@@ -188,7 +188,9 @@ def generate_api_token(self):
     generate_api_token.alters_data = True
 
     def generate_scratch_codes(self):
-        codes = [pyotp.random_base32(length=16) for i in range(settings.DMOJ_SCRATCH_CODES_COUNT)]
+        def generate_scratch_code():
+            return "".join(secrets.choice(string.ascii_uppercase) for _ in range(16))
+        codes = [generate_scratch_code() for i in range(settings.DMOJ_SCRATCH_CODES_COUNT)]
```

Comment:
Nit: I know this was like this originally, but s/i/_ for consistency with the new code.

Reply:
Fixed.